### PR TITLE
fix: change originalItem from being static

### PR DESCRIPTION
**Pull request by tgandrews**
### Summary:
Original item was being held in `$__` which was located on the prototype
of the model. This meant that this value was being shared for all
instances of model.

To resolve this it is now stored on the model instance. To ensure that
it is not exposed to the user other than through `originalItem` it is
stored as a Symbol.

There is a test recreating this scenario in this PR. However our first recreation of this bug can be seen with this linked script. 

https://gist.github.com/tgandrews/a79cc76fca248c25a4ff9c5396166baf

Before this change:
```
Creating cats...
...done
List cats
...done
Read cats...
id: 2 originalId: 4
id: 9 originalId: 4
id: 1 originalId: 4
id: 3 originalId: 4
id: 5 originalId: 4
id: 8 originalId: 4
id: 7 originalId: 4
id: 0 originalId: 4
id: 6 originalId: 4
id: 4 originalId: 4
```
After this change
```
Creating cats...
...done
List cats
...done
Read cats...
id: 2 originalId: 2
id: 9 originalId: 9
id: 1 originalId: 1
id: 3 originalId: 3
id: 5 originalId: 5
id: 8 originalId: 8
id: 7 originalId: 7
id: 0 originalId: 0
id: 6 originalId: 6
id: 4 originalId: 4
```

### Is this ready to be merged into Dynamoose? (select 1):
- [x] Yes
- [ ] No

### Other:
- [x] I have searched through the GitHub pull requests to ensure this PR has not already been submitted
- [ ] I have updated the Dynamoose documentation (if required) given the changes I made
- [x] I have added/updated the Dynamoose test cases (if required) given the changes I made
- [x] I agree that all changes made in this pull request may be distributed and are made available in accordance with the [Dynamoose license](https://github.com/dynamoosejs/dynamoose/blob/master/LICENSE.txt)
- [x] All of my commits and commit messages are detailed, explain what changes were made, and are easy to follow and understand
- [x] I have filled out all fields above

### CODE REVIEW

File: test/Model.js [hunk 1]
```diff
@@ -3268,6 +3268,35 @@ describe('Model', function () {
     });
   });
 
+  it('Should allow for originalItem on multiple models', (done) => {
```
Comment (fishcharlie) [inline test/Model.js]: This test fails without your changes you made, correct?

**Comment by fishcharlie** [comment]
@tgandrews, in PR #602 @Tirke made some changes to this logic. Do you both wanna discuss how to move forward with getting these changes in here and cross review the PRs?

File: test/Model.js (hunk 1 above)
Comment (tgandrews) [reply test/Model.js]: Correct

**Comment by tgandrews** [comment]
@fishcharlie are you happy for me to up the line limit for linting?

**Comment by fishcharlie** [comment]
@tgandrews 

1. Yes that is fine
2. Sounds good

**Comment by tgandrews** [comment]
@fishcharlie all resolved now. Sorry for the delay in resolving it but thanks for looking at it.

**Pull request merged**